### PR TITLE
Libraries: Create LibVideo library for parsing, decoding, and displaying videos

### PR DESCRIPTION
**Pull request by FalseHonesty**
This PR adds a new library, LibVideo, to handle parsing and decoding video-related content. This PR is still in its infancy and has a lot of work to do, but as of now, it contains a Matroska container parser and the start of a VP9 parser and decoder. I decided to start this PR as a draft sooner rather than later so I can incorporate feedback before I get too far.

At the time of creating this PR, I have implemented most of the [VP9 decoding spec](https://storage.googleapis.com/downloads.webmproject.org/docs/vp9/vp9-bitstream-specification-v0.6-20160331-draft.pdf) from section 6.0.0 -> 6.4.2, as well as a lot of other random (i.e. non-6.x.x) sections that were required for the above sections. This means that there is still a load of the spec left to implement.

I chose the Matroska container format and VP9 video codec to start LibVideo with because they are both open and royalty-free standards, and they both have pretty good specifications. Also, YouTube primarily uses WEBM files (which are matroska containers) and the VP9 codec ;)

I haven't contributed to Serenity in well over 6 months at this point, so I might be out of date on some more recent project developments or standards, so please let me know anything you find!

### CODE REVIEW

File: Userland/Libraries/LibVideo/VP9/Decoder.cpp [hunk 1]
```diff
@@ -0,0 +1,861 @@
...
+{
+    auto min_log_2 = 0u;
+    while ((u8)(MAX_TILE_WIDTH_B64 << min_log_2) < m_sb64_cols)
+        min_log_2++;
+    return min_log_2;
+}
+
+u16 Decoder::calc_max_log2_tile_cols()
+{
+    auto max_log_2 = 1;
```
Comment (Lubrsi) [inline Userland/Libraries/LibVideo/VP9/Decoder.cpp:420]:
```suggestion
    auto max_log_2 = 1u;
```

File: Userland/Libraries/LibVideo/VP9/BitStream.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+bool BitStream::read_bool(u8 probability)
+{
+    auto split = 1u + (((m_bool_range - 1u) * probability) >> 8u);
+    bool return_bool;
+
+    if (m_bool_value < split) {
+        m_bool_range = split;
+        return_bool = false;
+    } else {
+        m_bool_value -= split;
```
Comment (Lubrsi) [inline Userland/Libraries/LibVideo/VP9/BitStream.cpp:129]:
```suggestion
        m_bool_range -= split;
        m_bool_value -= split;
```

File: Userland/Libraries/LibVideo/VP9/Decoder.cpp [hunk 3]
```diff
@@ -0,0 +1,861 @@
...
+    // AboveNonzeroContext[0..2][0..MiCols*2-1] = 0
+    // AboveSegPredContext[0..MiCols-1] = 0
+    // AbovePartitionContext[0..Sb64Cols*8-1] = 0
+    return true;
+}
+
+u32 Decoder::get_tile_offset(u32 tile_num, u32 mis, u32 tile_size_log2)
+{
+    u32 super_blocks = (mis + 7) >> 3u;
+    u32 offset = ((tile_num * super_blocks) >> tile_size_log2) << 3;
```
Comment (Lubrsi) [inline Userland/Libraries/LibVideo/VP9/Decoder.cpp:802]:
```suggestion
    u32 offset = ((tile_num * super_blocks) >> tile_size_log2) << 3u;
```

**Review by Lubrsi — COMMENTED** [review]
Excellent work :). I've had a look through the decoder and noted things that may need changing.

**Comment by FalseHonesty** [comment]
> Excellent work :). I've had a look through the decoder and noted things that may need changing.

Wow, very tricky bugs to spot, thank you so much for your help!

**Comment by FalseHonesty** [comment]
I don't foresee having time to continue working on this PR, but it's a great jumping off point for anyone wishing to pick it up later, if they so choose.

**Pull request closed without merging**